src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -205,6 +206,31 @@ public static string GetPathRoot(string path)
 
         private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
         {
+            if (Environment.IsMac)
+            {
+                GetCryptoRandomBytesApple(bytes, byteCount);
+            }
+            else
+            {
+                GetCryptoRandomBytesOpenSsl(bytes, byteCount);
+            }
+        }
+
+        private static unsafe void GetCryptoRandomBytesApple(byte* bytes, int byteCount)
+        {
+            Debug.Assert(bytes != null);
+            Debug.Assert(byteCount >= 0);
+
+            int result = CCRandomGenerateBytes(bytes, byteCount);
```

Comment:
Are we going to have any trouble with this with AOT eventually, e.g. a P/Invoke to something that may not exist?